analyzer/src/funTest/kotlin/PubTest.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.analyzer
+
+import com.here.ort.analyzer.managers.Pub
+import com.here.ort.downloader.VersionControlSystem
+import com.here.ort.model.config.AnalyzerConfiguration
+import com.here.ort.model.yamlMapper
+import com.here.ort.utils.normalizeVcsUrl
+import com.here.ort.utils.test.DEFAULT_ANALYZER_CONFIGURATION
+import com.here.ort.utils.test.DEFAULT_REPOSITORY_CONFIGURATION
+import com.here.ort.utils.test.USER_DIR
+import com.here.ort.utils.test.patchExpectedResult
+import io.kotlintest.shouldBe
+import io.kotlintest.matchers.startWith
+import io.kotlintest.shouldNotBe
+import io.kotlintest.specs.WordSpec
+
+import java.io.File
+
+class PubTest : WordSpec() {
+    private val projectsDir = File("src/funTest/assets/projects/synthetic/pub/").absoluteFile
+    private val projectsDirExternal = File("src/funTest/assets/projects/external/").absoluteFile
+    private val vcsDir = VersionControlSystem.forDirectory(projectsDir)!!
+    private val vcsUrl = vcsDir.getRemoteUrl()
+    private val vcsRevision = vcsDir.getRevision()
+
+    init {
+
+        "Pub" should {
+            "resolve dart http dependencies correctly" {
+                val workingDir = File(projectsDirExternal, "http")
+                val packageFile = File(workingDir, "pubspec.yaml")
+                var expectedResultFile = File(projectsDirExternal, "http-expected-output.yml");
+
+                val result = createPubForExternal().resolveDependencies(listOf(packageFile))[packageFile]
+                result!!.errors.size shouldBe 0
```

Comment:
This check is implicitly done by the comparison below.

Reply:
done